### PR TITLE
fixed php 7.2  Warning: count(): Parameter must be an array or an obj…

### PR DESCRIPTION
**Pull request by firma**
…ect that implements Countable

fixed php 7.2 ,Warning: count(): Parameter must be an array or an object that implements Countable

### CODE REVIEW

**Comment by firma** [comment]
fixed php 7.2  Warning: count(): Parameter must be an array or an object that implements Countable

**Comment by GrahamCampbell** [comment]
👎 Why is this ever null. Instead, the place that incorrectly sets it to null should be fixed.

**Comment by lyyhd** [comment]
it has fix?

File: src/Handler/CurlFactory.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ public function release(EasyHandle $easy)
         $resource = $easy->handle;
         unset($easy->handle);
 
-        if (count($this->handles) >= $this->maxHandles) {
+        if (is_array($this-handles) && count($this->handles) >= $this->maxHandles) {
```
Comment (marekr) [inline src/Handler/CurlFactory.php]: missing a > btw

**Comment by sagikazarmark** [comment]
After a quick look: I don't see any code paths that would end up in `$handles` being null.

**Comment by marekr** [comment]
I have the same error in my codebase....I'll take a look at what exactly is failing.

Mind you, the fix has nothing to do with a null check. PHP 7.2 enforces count() only being usable on ICountable. Previously if you passed random variable type it would return whatever flawed value it did.

**Comment by pfrenssen** [comment]
This has been fixed in #1686. This can be closed.

File: src/Handler/CurlFactory.php (hunk 1 above)
Comment (PierreMaurel) [inline src/Handler/CurlFactory.php]: Missing > in this - handeles

Comment (yassinechabli) [reply src/Handler/CurlFactory.php]: the count function does support null in php 7.2 .

Comment (fja3omega) [reply src/Handler/CurlFactory.php]: This works for js_composer

Comment (songwutk) [reply src/Handler/CurlFactory.php]: is_countable($this->handles)

**Comment by GrahamCampbell** [comment]
Ping.

**Comment by dj12342** [comment]
@extends('layouts.app')

 

@section('content')

        <h1> {{$title}}</h1>

        @if(count($Services > 1))

         <ul>

           @foreach($Services as $Services)

                <li>{{$Services}}</li>

           @endforeach

        </ul>

    @endif

@endsection

**Comment by BestOlumese** [comment]
@extends('layouts.app')

@section('content')

    <h1> {{$title}}</h1>

    @if(count($Services > 1))

     <ul>

       @foreach($Services as $Services)

            <li>{{$Services}}</li>

       @endforeach

    </ul>

@endif

**Pull request closed without merging**